File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/calculatetaskduration.rb
module Asposetasksjava
  module CalculateTaskDuration
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        tsk = Rjb::import('com.aspose.tasks.Tsk')
        time_unit_type = Rjb::import('com.aspose.tasks.TimeUnitType')

        # Get a task to calculate its duration in different formats
		task = project.getRootTask().getChildren().getById(1)

		# Get the duration in Minutes
		puts task.get(tsk.DURATION).convert(time_unit_type.Minute).toDouble()

		# Get the duration in Days
		puts task.get(tsk.DURATION).convert(time_unit_type.Day).toDouble()
		
		# Get the duration in Hours
		puts task.get(tsk.DURATION).convert(time_unit_type.Hour).toDouble()
		
		# Get the duration in Weeks
		puts task.get(tsk.DURATION).convert(time_unit_type.Week).toDouble()
		
		# Get the duration in Months
		puts task.get(tsk.DURATION).convert(time_unit_type.Month).toDouble()
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/CalculateTaskDuration.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Tsk as Tsk;
use com\aspose\tasks\TimeUnitType as TimeUnitType;
class CalculateTaskDuration{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $tsk = new Tsk();
        $time_unit_type = new TimeUnitType();

        # Get a task to calculate its duration in different formats
		$task = $project->getRootTask()->getChildren()->getById(1);

		# Get the duration in Minutes
		print (double)(string)($task->get($tsk->DURATION)->convert($time_unit_type->Minute)).PHP_EOL;//.toDouble();

		# Get the duration in Days
        print (double)(string)($task->get($tsk->DURATION)->convert($time_unit_type->Day)).PHP_EOL;//.toDouble());

		# Get the duration in Hours
        print (double)(string)($task->get($tsk->DURATION)->convert($time_unit_type->Hour)).PHP_EOL;//.toDouble();

		# Get the duration in Weeks
        print (double)(string)($task->get($tsk->DURATION)->convert($time_unit_type->Week)).PHP_EOL;//.toDouble();

		# Get the duration in Months
        print (double)(string)($task->get($tsk->DURATION)->convert($time_unit_type->Month)).PHP_EOL;//.toDouble();

    }
}
?>